scarb/src/compiler/db.rs

```diff
@@ -134,12 +136,39 @@ fn build_project_config(unit: &CairoCompilationUnit) -> Result<ProjectConfig> {
         .map(|component| {
             let experimental_features = component.package.manifest.experimental_features.clone();
             let experimental_features = experimental_features.unwrap_or_default();
+            // Those are direct dependencies of the component.
+            let dependencies_summary: Vec<&ManifestDependency> = component
+                .package
+                .manifest
+                .summary
+                .full_dependencies()
+                .collect();
+
+            // We iterate over all of the compilation unit components to get dependency's version.
+            let dependencies = unit
+                .components
+                .iter()
+                .filter(|component_as_dependency| {
```

Comment:
Have you verified that we don't need to add the component itself as the dep? (i.e. import stuff as `use package_name::...` in said package) Because it would not pop up in `full_dependencies`, right? 🤔

Reply:
Right, will change that